rusk-wallet/src/lib/prompt.rs

```diff
@@ -471,17 +471,16 @@ fn check_valid_denom(num: f64, balance: f64) -> Result<(), String> {
 }
 
 /// Request amount of tokens
-fn request_token_amt(action: &str, balance: f64) -> Dusk {
+fn request_token_amt(action: &str, balance: f64) -> f64 {
```

Comment:
There is something off about this change. `f64` should be the unit of measure of `Dusk`. This means if `Dusk` isn't able to receive `f64` (and it isn't, being `u64`), its type definition is wrong and should be changed to `f64`

Reply:
@ureeves will probably want to jump into this discussion

Reply:
Dusk isn't an `f64`. If anything it's a fixed point number in with a minimum increment of `1*10⁻⁹` (meaning one LUX). `u64` is actually more convenient to represent this in memory since it also allows us to pass Dusk through the stack without having to convert it.

The way we should think about it is a that a subset of `f64` can be used as a *representation* of Dusk. This is tricky, and in the future I would probably vote for `Dusk` being a formal struct instead of a type alias.

That said, I still don't understand why we need this function to return `f64` instead of Dusk.

Reply:
I don't think we can use plain `f64` here as units of `Dusk`. The `f64` by itself doesn't mean anything and might be confusing.

What we can do is create two structures `Dusk` and `Lux`, and implement the normal arithmetic and conversions between them

- `core::ops::{Add, Sub, Mul, Div}`
- `From`

That way, I just `Lux::from(dusk)` or `lux + dusk`

Reply:
I created a follow-up issue for this: #675